algopy_testing/src/algopy_testing/arc4.py

```diff
@@ -59,6 +58,55 @@ def from_log(cls, log: algopy.Bytes, /) -> typing.Self:
         ...
 
 
+class String(_ABIEncoded):
+    """An ARC4 sequence of bytes containing a UTF8 string"""
+
+    _value: bytes
+
+    def __init__(self, value: algopy.String | str = "", /) -> None:
+        value = as_string(value)
+        bytes_value = value.encode("utf-8")
+        self._value = len(bytes_value).to_bytes(_ABI_LENGTH_SIZE) + bytes_value
+
+    @property
+    def native(self) -> algopy.String:
+        """Return the String representation of the UTF8 string after ARC4 decoding"""
+        return algopy.String(self._value[_ABI_LENGTH_SIZE:].decode("utf-8"))
```

Comment:
Again to preserve semantic compatibility here this should avoid doing the decode operation, and instead construct an `algopy.String` using just the bytes

Reply:
It has been fixed in this commit daab1b78c560ff747c4569d4f44321ad180e70f7.